Assets/Scripts/Gameplay/GameplayObjects/Character/ClientCharacterVisualization.cs

```diff
@@ -155,6 +186,21 @@ public override void OnNetworkDespawn()
             enabled = false;
         }
 
+        IEnumerator KilledDestroyProcess()
+        {
+            yield return new WaitForSeconds(m_KilledDestroyDelaySeconds);
+
+            Hide();
```

Comment:
```suggestion
            if (NetworkObject.IsSceneObject == false)
            {
                NetworkObject.Despawn(true);
            }
            else
            {
                Hide();
            }
```

This would allow dynamically-spawned imps to still be destroyed so that we don't end up with too many instances of them and so it doesn't prevent portals to continue spawning them